target/linux/ipq40xx/image/Makefile

```diff
@@ -627,6 +627,35 @@ define Device/qcom_ap-dk04.1-c1
 endef
 TARGET_DEVICES += qcom_ap-dk04.1-c1
 
+define Device/p2w_r619ac
+	$(call Device/FitzImage)
+	$(call Device/UbiFit)
+	DEVICE_VENDOR := P&W
+	DEVICE_MODEL := R619AC
+	SOC := qcom-ipq4019
+	DEVICE_DTS_CONFIG := config@10
+	BLOCKSIZE := 128k
+	PAGESIZE := 2048
+	IMAGES += nand-factory.bin
+	IMAGE/nand-factory.bin := append-ubi | qsdk-ipq-factory-nand
+	DEVICE_PACKAGES := ipq-wifi-p2w_r619ac
+endef
+TARGET_DEVICES += p2w_r619ac
```

Comment:
Hm, from the specs you provided this one does not have NAND at all.
So what's with the UbiFit, block sizes, etc?

Reply:
These two devices are actually the same one which has a 16MiB spi nor flash and a 128MiB nand flash. The stock firmware used only 64MiB and left 64MiB free space on the nand. 
The p2w_r619ac uses the same partition layout with the stock firmware while the p2w_r619ac-128m uses the full 128MiB space on the nand.

Reply:
Wait, in the PR specs, the first model has no NAND listed while the second one has?
Whats correct?

Reply:
@robimarko See my comment below.

Reply:
> Wait, in the PR specs, the first model has no NAND listed while the second one has?
> Whats correct?

product link: 
https://www.amazon.co.uk/dp/B07DL1PJ41/ref=sr_1_fkmr0_1?ie=UTF8&qid=1528790821&sr=8-1-fkmr0&keywords=P%26W+R619AC+802.11AC+wave2+Smart+wifi+router+Dual-band+Gigabit+with+Qualcomm+ARM+4+core
product spec in chinese:
http://www.p2w.net.cn/skin/text/R619AC%201300M%E4%BC%81%E4%B8%9A%E7%BA%A7%E5%85%A8%E5%8D%83%E5%85%86%E6%97%A0%E7%BA%BF%E8%B7%AF%E7%94%B1%E5%99%A8%E4%BA%A7%E5%93%81%E8%A7%84%E6%A0%BC%E4%B9%A6-V1.2.pdf

Reply:
Ok, so they both have NAND onboard.
I was confused as to why are NAND only specifics defined for both devices as one listed only NOR

Reply:
The commit message may be wrong.